domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+   * @param array $ids
+   *   List of entity IDs for the selected domain and all of type $entity_type.
+   *
+   * @return int
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function reassignEntities(string $entity_type, string $field, DomainInterface $old_domain, DomainInterface $new_domain, $ids) {
+    $entity_storage = \Drupal::entityTypeManager()->getStorage($entity_type);
+
+    // @TODO is there a problem loading many, possibly big, entities in one go?
+    $entities = $entity_storage->loadMultiple($ids);
+
+    // @TODO do we need to protect against $entity not having the field $field?
+    /** @var \Drupal\domain\DomainInterface $entity */
+    foreach($entities as $entity) {
+      $changed = FALSE;
+      // Multivalue fields are used, so check each one.
+      foreach ($entity->get($field) as $k => $item) {
+        // NB: NOT strict ==
+        if ($item->target_id == $old_domain->id()) {
+
+          if ($this->is_dry_run) {
+            $this->logger()->info(dt('DRYRUN: Update domain membership for entity @id to @new.',
+              [ '@id' => $entity->id(), '@new' => $new_domain->id() ]));
+
+            // Don't set changed, so don't save either.
+            continue;
+          }
+
+          $changed = TRUE;
+          $item->target_id = $new_domain->id();
+        }
+      }
+      if ($changed) {
+        $entity->save();
+      }
+    }
+    return count($entities);
+  }
+
+  /**
+   * Return the Domain object corresponding to a policy string.
+   *
+   * @param string $policy
+   *   In general one of 'prompt' | 'default' | 'ignore' or a domain entity
+   *   machine name, but this function does not process 'prompt'.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface|\Drupal\domain\DomainInterface|null
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  protected function getDomainInstanceFromPolicy(string $policy) {
+    $domain_storage = $this->getStorage();
+    switch($policy) {
+      case 'default':
+        $new_domain = $domain_storage->loadDefaultDomain();
+        break;
+
+      case 'prompt':
+      case 'ignore':
+        return NULL;
+
+      default:
+        $new_domain = $domain_storage->load($policy);
+        break;
+    }
+    return $new_domain;
+  }
+
+  /**
+   * Reassign entities of the supplied type to the $policy domain.
+   *
+   *  $options = [
+   *    'entity_filter' => 'node',
+   *    'policy' => 'prompt' | 'default' | 'ignore'
+   *    'multidomain' => FALSE,
+   *  ];
+   *
+   * @param DomainInterface[] $domains
+   *   List of the domains to reassign content away from.
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  protected function reassignLinkedEntities($domains, array $options = ['chatty' => null, 'policy' => null]) {
+    $entity_typenames = $this->findDomainEnabledEntities();
+    // DOMAIN_ADMIN_FIELD too??
```

Comment:
DOMAIN_ADMIN_FIELD would be a new operation.

Reply:
Not really understanding. Should I remove the comment?